docs/book/user-guide/starter-guide/manage-artifacts.md

```diff
@@ -301,6 +301,40 @@ Even if an artifact is created externally, it can be treated like any other arti
 It is also possible to use these functions inside your ZenML steps. However, it is usually cleaner to return the artifacts as outputs of your step to save them, or to use External Artifacts to load them instead.
 {% endhint %}
 
+## Linking data **not** produced by ZenML pipelines as a ZenML artifact
+
+Sometimes, data is produced completely outside of ZenML and can be conveniently store on a given storage. A good example of this is the checkpoint files produced as a side-effect of the Deep Learning model training. We know that the intermediate data of the deep learning frameworks is quite big and there is no good reason to move it around again and again, if it can be produced directly in the artifact store boundaries and later just linked to become an artifact of ZenML.
+
+```python
+# Here we explore the Pytorch Lightning example
+# to fit model and store the checkpoints on remote
+# location.
+
+# The important assumption is that active artifact
+# store is s3://my_bucket, otherwise the linking
+# going to fail with an error.
+
+from zenml.client import Client
+from zenml import link_folder_as_artifact
+from pytorch_lightning import Trainer
+
+# Define the model and fit it
+model = ...
+trainer = Trainer(default_root_dir="s3://my_bucket/my_model_data/")
+try:
+    trainer.fit(model)
+finally:
+    # We now link those checkpoints in ZenML as an artifact
+    # This will create a new artifact version
+    link_folder_as_artifact(folder_uri="s3://my_bucket/my_model_data/ckpts", name="my_model_ckpts")
```

Comment:
cant we also have callbacks to do this one by one in case training gets interrupted? might make sense to have `link_file_as_artifact` for that

Reply:
How will versioning work in this case?

Reply:
In this case, the versioning would not work too much, but I added an extra example to this section.
In fact, `link_file_as_artifact` would mean the need to move data around, since our materialized relies on folders, AFAIK. I would wait for @schustmi to confirm on Monday, but even with the current approach on folders, I'm rather satisfied with how this experience could look like under a Callback of Lightning directly.

Reply:
I'm guessing the callback will create files with the epoch name in the checkpoints directory? In that case, the repeated calls to this method would create artifact versions, but they would all point to the same directory which will have all the checkpoint files inside?

Currently when ZenML creates an artifact, we usually point the path to a directory. However we might be able to workaround that by simply creating an artifact which points to a specific file, as long as the materializer handles it correctly? Not sure if there are additional checks and whether we get the exact checkpoint file URL inside the handler, but worth a try I guess.

If that's not possible, it's somehow not that useful that actually call that at the end of every epoch, right? Just doing it once before the training call would actually be good enough I assume?

Reply:
I depicted how this shall be used in the example in docs, but you are all right - pointing to a file is more intuitive for the checkpointing case (and for others maybe). I will try to switch gears here and point to the file instead.

Reply:
I made it available for single files as well